CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -668,119 +716,262 @@ function Test-VRouterAgentIntegration {
-            $Container1Name = "jolly-lumberjack"
-            $Container2Name = "juniper-tree"
+    function Test-TcpTrafficSingleNet {
+        Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
 
-            $CreateContainer1Success = Create-ContainerInRemoteSession `
-                -Session $Session `
-                -NetworkName $Network1Name `
-                -ContainerName $Container1Name
-            $CreateContainer2Success = Create-ContainerInRemoteSession `
-                -Session $Session `
-                -NetworkName $Network2Name `
-                -ContainerName $Container2Name
+        $Name = $MyInvocation.MyCommand.Name
 
-            if ($CreateContainer1Success -ne 0 -or $CreateContainer2Success -ne 0) {
-                throw "Container creation failed. EXPECTED: succeeded."
-            }
+        $Job.StepQuiet($Name, {
+            Test-Tcp `
+                -Name $Name `
+                -Session1 $Session `
+                -Session2 $Session `
+                -TestConfiguration $TestConfiguration `
+                -Network2 $TestConfiguration.DockerDriverConfiguration.TenantConfiguration.NetworkWithPolicy1
+        })
+    }
 
-            $Container2IP = Invoke-Command -Session $Session -ScriptBlock {
-                & docker exec $Using:Container2Name powershell -Command "(Get-NetAdapter | Select-Object -First 1 | Get-NetIPAddress).IPv4Address"
+    # Test-TcpTrafficMultipleNets doesn't actually connect (only flow is created)
+
+    function Test-MultihostTcpTrafficSingleNet {
+        Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session1,
+               [Parameter(Mandatory = $true)] [PSSessionT] $Session2,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        $Name = $MyInvocation.MyCommand.Name
+
+        $Job.StepQuiet($Name, {
+            Test-Tcp `
+                -Name $Name `
+                -Session1 $Session1 `
+                -Session2 $Session2 `
+                -TestConfiguration $TestConfiguration `
+                -Network2 $TestConfiguration.DockerDriverConfiguration.TenantConfiguration.NetworkWithPolicy1
+         })
+    }
+
+    # Test-MultihostTcpTrafficMultipleNets results in bluescreen on $Session1
+
+    function Test-Tcp {
+        Param ([Parameter(Mandatory = $true)] [String] $Name,
+               [Parameter(Mandatory = $true)] [PSSessionT] $Session1,
+               [Parameter(Mandatory = $true)] [PSSessionT] $Session2,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration,
+               [Parameter(Mandatory = $false)] [NetworkConfiguration] $Network1,
+               [Parameter(Mandatory = $false)] [NetworkConfiguration] $Network2,
+               [Parameter(Mandatory = $false)] [Bool] $TestFlowInjection = $true,
```

Comment:
I don't think it is a good approach to merge those 3 tests in one method.

Reply:
Alternatives are:
* massive copy-paste for each variant of TCP test
* Make TCP test a class with methods like `.StartServer()`, `.TestFlows()`, etc... -- is it a good idea?

Reply:
Second option looks promising